code/modules/projectiles/guns/misc/medbeam.dm

```diff
@@ -52,15 +63,14 @@
 
 	current_target = target
 	active = TRUE
-	current_beam = new(user,current_target,time=6000,beam_icon_state="medbeam",btype=/obj/effect/ebeam/medical)
-	INVOKE_ASYNC(current_beam, /datum/beam.proc/Start)
+	current_beam = user.Beam(current_target, icon_state="medbeam", maxdistance = max_range, beam_type = /obj/effect/ebeam/medical)
+	RegisterSignal(current_beam, list(COMSIG_PARENT_QDELETING), .proc/beam_died)//this is a WAY better rangecheck than what was done before (process check)
+	QDEL_IN(current_beam, 10 MINUTES)
 
 	SSblackbox.record_feedback("tally", "gun_fired", 1, type)
 
 /obj/item/gun/medbeam/process()
-
-	var/source = loc
-	if(!mounted && !isliving(source))
+	if(!mounted && !isliving(loc))
```

Comment:
Does loc change throughout this process? was source needed?

Reply:
I checked, seemingly no and even if so the los_check proc gets a new turf to work off of. medbeam's loc is a person, it picks the loc's turf after that. doubt source was needed